controllers/resource_helper.go

```diff
@@ -61,111 +109,428 @@ func buildContainerSpec(instance *llamav1alpha1.LlamaStackDistribution, image st
+	if len(instance.Spec.Server.ContainerSpec.Command) > 0 {
+		container.Command = instance.Spec.Server.ContainerSpec.Command
+	}
+
+	if len(instance.Spec.Server.ContainerSpec.Args) > 0 {
+		container.Args = instance.Spec.Server.ContainerSpec.Args
+	}
+}
+
+// getMountPath returns the mount path, using custom path if specified.
+func getMountPath(instance *llamav1alpha1.LlamaStackDistribution) string {
+	if instance.Spec.Server.Storage != nil && instance.Spec.Server.Storage.MountPath != "" {
+		return instance.Spec.Server.Storage.MountPath
+	}
+	return llamav1alpha1.DefaultMountPath
+}
+
+// addStorageVolumeMount adds the storage volume mount to the container.
+func addStorageVolumeMount(instance *llamav1alpha1.LlamaStackDistribution, container *corev1.Container) {
+	mountPath := getMountPath(instance)
 	container.VolumeMounts = append(container.VolumeMounts, corev1.VolumeMount{
 		Name:      "lls-storage",
 		MountPath: mountPath,
 	})
+}
 
-	// Add ConfigMap volume mount if user config is specified
+// addUserConfigVolumeMount adds the user config volume mount to the container if specified.
+func addUserConfigVolumeMount(instance *llamav1alpha1.LlamaStackDistribution, container *corev1.Container) {
 	if instance.Spec.Server.UserConfig != nil && instance.Spec.Server.UserConfig.ConfigMapName != "" {
 		container.VolumeMounts = append(container.VolumeMounts, corev1.VolumeMount{
 			Name:      "user-config",
 			MountPath: "/etc/llama-stack/",
 			ReadOnly:  true,
 		})
+	}
+}
 
-		// Override the container entrypoint to use the custom config file instead of the default template
-		container.Command = []string{"python", "-m", "llama_stack.distribution.server.server"}
-		container.Args = []string{"--config", "/etc/llama-stack/run.yaml"}
+// addCABundleVolumeMount adds the CA bundle volume mount to the container if TLS config is specified.
+// For multiple keys: the init container writes ca-bundle.crt to the root of the emptyDir volume,
+// and the main container mounts it with SubPath to /etc/ssl/certs/ca-bundle.crt.
+// For single key: the main container directly mounts the ConfigMap key.
+// Also handles auto-detected ODH trusted CA bundle ConfigMaps.
+func addCABundleVolumeMount(ctx context.Context, r *LlamaStackDistributionReconciler, instance *llamav1alpha1.LlamaStackDistribution, container *corev1.Container) {
+	if instance.Spec.Server.TLSConfig != nil && instance.Spec.Server.TLSConfig.CABundle != nil {
+		caBundleKey := getCABundleKey(instance.Spec.Server.TLSConfig.CABundle)
+
+		container.VolumeMounts = append(container.VolumeMounts, corev1.VolumeMount{
+			Name:      "ca-bundle",
+			MountPath: "/etc/ssl/certs/" + caBundleKey,
+			SubPath:   caBundleKey,
+			ReadOnly:  true,
+		})
+	} else if r != nil {
+		// Check for auto-detected ODH trusted CA bundle
+		if _, keys, err := r.detectODHTrustedCABundle(ctx, instance); err == nil && len(keys) > 0 {
+			// Mount the auto-detected consolidated CA bundle
```

Comment:
Shouldn't all the cert keys be consolidated in one single file, `SSL_CERT_FILE`? While testing, I see its not the case

Reply:
Interesting, I was able to confirm that it does seem there has been a regression since opening this PR. I haven't tracked down the issue yet, I'll continue troubleshooting and see what's causing it.

Reply:
Ok, the concatenation bug should now be resolved. I've updated the PR with the latest changes from my branch, and the tests are passing.